src/WebJobs.Script/Config/ActiveHostConfigurationSource.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    public class ActiveHostConfigurationSource : IConfigurationSource
+    {
+        public ActiveHostConfigurationSource(IServiceProvider serviceProvider)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        public IServiceProvider ServiceProvider { get; set; }
+
+        public IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            return new ActiveHostConfigurationProvider(this);
+        }
+
+        private class ActiveHostConfigurationProvider : ConfigurationProvider
+        {
+            private readonly ActiveHostConfigurationSource _configurationSource;
+
+            public ActiveHostConfigurationProvider(ActiveHostConfigurationSource configurationSource)
+            {
+                _configurationSource = configurationSource ?? throw new ArgumentNullException(nameof(configurationSource));
+            }
+
+            public override bool TryGet(string key, out string value)
+            {
+                if (_configurationSource?.ServiceProvider?.GetService(typeof(IConfiguration)) is IConfigurationRoot activeHostConfiguration)
+                {
+                    return
+                        (value = activeHostConfiguration.GetValue<string>(key)) != null ||
+                        (value = activeHostConfiguration.GetValue<string>(NormalizeKey(key))) != null;
+                }
+
+                value = default;
+                return false;
+            }
+
+            public override IEnumerable<string> GetChildKeys(IEnumerable<string> earlierKeys, string parentPath)
+            {
+                var activeHostConfiguration = _configurationSource?.ServiceProvider?.GetService(typeof(IConfiguration)) as IConfigurationRoot;
+                var keys = new HashSet<string>();
```

Comment:
No need to create an instance here.

Instantiate if the configuration is not `null`, otherwise; return an `Enumerable.Empty<string>()`

Reply:
If activeHostConfiguration == null, shouldn't it still return earlierKeys? That's how I interpreted this: https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.iconfigurationprovider.getchildkeys?view=dotnet-plat-ext-5.0#Microsoft_Extensions_Configuration_IConfigurationProvider_GetChildKeys_System_Collections_Generic_IEnumerable_System_String__System_String_